### ISSUE TITLE
Create a WithDsl for constructing/configuring the entire system

### ISSUE THREAD

**Issue by osoykan** [Trendyol/stove4k#41]
MongoDB has this implementation; use the same approach in other places, too.

Mongo: https://github.com/Trendyol/stove4k/blob/main/lib/stove-testing-e2e-mongodb/src/main/kotlin/com/trendyol/stove/testing/e2e/mongodb/MongodbOptionsDsl.kt

Usage:
https://github.com/Trendyol/stove4k/blob/78966d89c8fab6b5022aa3cc518f1e14879062a0/lib/stove-testing-e2e-mongodb/src/test/kotlin/com/trendyol/stove/testing/e2e/mongodb/MongodbTestSystemTests.kt#L19-L26

**Comment by osoykan** [comment]
Example usage:

```kotlin
TestSystem(baseUrl = "http://localhost:8001")
    .with {
        httpClient()
        couchbase { CouchbaseSystemOptions("Stove") }
        kafka()
        wiremock {
            WireMockSystemOptions(
                port = 9099,
                removeStubAfterRequestMatched = true,
                afterRequest = { e, _, _ ->
                    logger.info(e.request.toString())
                }
            )
        }
        springBoot(
            runner = { parameters ->
                stove.spring.example.run(parameters) { this.addTestS
            },
            withParameters = listOf(
                "server.port=8001",
                "logging.level.root=info",
                "logging.level.org.springframework.web=info",
                "spring.profiles.active=default",
                "kafka.heartbeatInSeconds=2",
                "kafka.autoCreateTopics=true",
                "kafka.offset=earliest",
                "kafka.secureKafka=false"
            )
        )
    }.run()
```

**Comment by osoykan** [comment]
Implemented in #88
